src/main/host/syscall/socket.c

```diff
@@ -366,8 +354,15 @@ static int _syscallhandler_setSocketOptHelper(SysCallHandler* sys, Socket* sock,
         case SO_SNDBUF: {
             const unsigned int* val =
                 process_getReadablePtr(sys->process, sys->thread, optvalPtr, sizeof(int));
-            size_t newsize =
-                (*val) * 2; // Linux kernel doubles this value upon setting
+            size_t newsize = (*val) * 2; // Linux kernel doubles this value upon setting
+
+            // Linux also has limits SOCK_MIN_SNDBUF (slightly greater than 4096) and the sysctl max
+            // limit. We choose some reasonable limits for Shadow. The minimum limit in man 7 socket
+            // is incorrect.
+            newsize = MAX(newsize, 4096);
+            // This upper limit was chosen as an arbitrarily high number.
```

Comment:
It's worth mentioning a bit of the reasoning. e.g. that this is effectively a ceiling of how many un-acked packets we can have, since we need to buffer the packets for possible retransmission (or was that a different setting?), and Rob's back-of-the-envelope math to ensure we can saturate a 1 Gbps link

Reply:
So we're not sure what Linux does, but in Shadow this setting doesn't affect the TCP retransmit queue as the socket queue and retransmit queue are separate. So I don't think there's actually a reason to keep it large, but am just doing so anyways to not break anything.

Reply:
Should we still have some upper limit to avoid excessive allocation/buffering? 256 MiB seems like a plausible limit to me; I was just looking for a comment with our current understanding of how to choose this limit.

If we do think no upper limit is needed here (e.g. because an upper limit set somewhere else will already have the desired effect), that'd probably also be worth a comment.

Reply:
I added it back with hopefully a better comment.